pkg/prometheus/promcfg.go

```diff
@@ -84,6 +86,7 @@ func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, e
 		version:                version,
 		prom:                   p,
 		endpointSliceSupported: endpointSliceSupported,
+		ctx:                    ctx,
```

Comment:
Its preferred to pass context as first argument to function than having it part of struct